<?xml version="1.0"?>
<launch>
    <arg name="topic_imu"           default="/imu" />
    <arg name="path_bag"            default="/home/$(env USER)/data/dataset_calibration/test-300-garage-fixed.bag" />
    <arg name="bag_start"           default="1" />
    <arg name="bag_durr"            default="600" />
    <arg name="scan4map"            default="15" />
    <arg name="lidar_model"         default="HDL_32E" />
    <arg name="ndtResolution"       default="0.5" /> <!-- 0.5 for indoor case and 1.0 for outdoor case -->

    <arg name="time_offset_padding" default="0.015" />
    <arg name="show_ui"    default="false" />
    <arg name="gyro_weight" default="28.0" />
    <arg name="accelerometer_weight" default="18.5" />
    <arg name="lidar_weight" default="10.0" />
    <arg name="initial_p_LinI" />
    <arg name="initial_q_LtoI" />

    <node pkg="li_calib" type="li_calib_gui" name="li_calib_gui" output="screen">
    <!-- <node pkg="li_calib" type="li_calib_gui" name="li_calib_gui" output="screen" clear_params="true" launch-prefix="gdb -ex run &#45;&#45;args">-->

        <param name="topic_imu"         type="string"   value="$(arg topic_imu)" />
        <param name="topic_lidar"       type="string"   value="/velodyne_packets" />
        <param name="LidarModel"        type="string"   value="$(arg lidar_model)" />
        <param name="path_bag"          type="string"   value="$(arg path_bag)" />
        <param name="bag_start"         type="double"   value="$(arg bag_start)" />
        <param name="bag_durr"          type="double"   value="$(arg bag_durr)" /> <!-- for data association -->
        <param name="scan4map"          type="double"   value="$(arg scan4map)" />
        <param name="ndtResolution"     type="double"   value="$(arg ndtResolution)" />
        <param name="gyro_weight"       type="double"   value="$(arg gyro_weight)" />
        <param name="accelerometer_weight"     type="double"   value="$(arg accelerometer_weight)" />
        <param name="lidar_weight"       type="double"   value="$(arg lidar_weight)" />
        <rosparam param="initial_p_LinI" subst_value="True">$(arg initial_p_LinI)</rosparam>
        <rosparam param="initial_q_LtoI" subst_value="True">$(arg initial_q_LtoI)</rosparam>

        <param name="time_offset_padding"   type="double"   value="$(arg time_offset_padding)" />
        <param name="show_ui"               type="bool"     value="$(arg show_ui)" />
    </node>

</launch>
